cpp/tests/io/fst/common.hpp

```diff
@@ -39,36 +40,46 @@ enum DFA_STATES : char {
   TT_NUM_STATES
 };
 
-// Definition of the symbol groups
-enum PDA_SG_ID {
-  OBC = 0U,          ///< Opening brace SG: {
-  OBT,               ///< Opening bracket SG: [
-  CBC,               ///< Closing brace SG: }
-  CBT,               ///< Closing bracket SG: ]
-  QTE,               ///< Quote character SG: "
-  ESC,               ///< Escape character SG: '\'
-  OTR,               ///< SG implicitly matching all other characters
+/**
+ * @brief Definition of the symbol groups
+ */
+enum class dfa_symbol_group_id : uint32_t {
+  OPENING_BRACE,     ///< Opening brace SG: {
+  OPENING_BRACKET,   ///< Opening bracket SG: [
+  CLOSING_BRACE,     ///< Closing brace SG: }
+  CLOSING_BRACKET,   ///< Closing bracket SG: ]
+  QUOTE_CHAR,        ///< Quote character SG: "
+  ESCAPE_CHAR,       ///< Escape character SG: '\'
+  OTHER_SYMBOLS,     ///< SG implicitly matching all other characters
   NUM_SYMBOL_GROUPS  ///< Total number of symbol groups
 };
 
+// Aliases for readability of the transition table
+constexpr auto TT_OOS = dfa_states::TT_OOS;
```

Comment:
FYI C++20 will make this easier with `using enum ...`. https://en.cppreference.com/w/cpp/language/enum#Using-enum-declaration

Reply:
Nice! Good to know. Thanks for pointing that out 💡